daml-assistant/BUILD.bazel

```diff
@@ -132,3 +132,16 @@ da_haskell_test(
         "//libs-haskell/da-hs-base",
     ],
 )
+
+# A good target for da-ghcid, it includes everything in the subtree.
+da_haskell_repl(
+    name = "repl",
```

Comment:
Maybe this should be the new default for `da-ghci` then?

Reply:
I'm in favor of that!

Reply:
In particular, making the default target be `:repl` instead of `//:damlc`.

Reply:
I think the number of people working on the assistant is far smaller than the number of people working on damlc so I don’t think swapping the default makes sense.

Reply:
@cocreature I think the point is to make the "default" depend on the working directory, so if you're on `//daml-assistant` you get `//daml-assistant:repl` but if you're in `//daml-foundations/daml-ghc` then you get the `:repl` rule for that subtree instead? And we can define a top-level `:repl` rule that includes basically everything.

Reply:
That sounds reasonable but ofc only once we actually have `:repl` rules in all subtrees.

Reply:
I'll add them now then. ;-)

Reply:
Or, nicer move, add a top-level `//:repl` that includes both damlc and daml-assistant, and set that as the default. That way I don't need to go touching a million bazel files and anyone can add targets they want later.

Reply:
Sounds reasonable. Not quite sure how `:main` works in that case but I’d be fine if that doesn’t work at all in this case.

Reply:
The thing with `main` is you can't have two `Main.main` functions being loaded in a single repl, so as long as all (but possibly one) target use `main_function` to rename their main funciton, then it works out. For this reason I renamed daml-helper's Main.main to DamlHelper.Main.main.